content/tutorials/presence.textile

```diff
@@ -137,110 +206,145 @@ blang[ruby].
+  ```[nodejs]
+    // Subscribe to the presence set to receive updates
+    await channel.presence.subscribe((presenceMessage) => {
+      console.log(
+        "Presence update: " +
+          presenceMessage.action +
+          " from " +
+          presenceMessage.clientId
+      );
+      // Update the list of channel members when the presence set changes
+      channel.presence.get((err, members) => {
+        if (err) {
+          return console.error(`Error retrieving presence data: ${err}`);
+        }
+        const clientIDs = members.map(({ clientId }) => clientId);
+        console.log("The presence set now consists of: ", clientIDs);
+      });
+    });
+  ```
 
 blang[ruby].
   ```[ruby]
-    channel.presence.subscribe do |presence_msg|
-      puts "Received a #{presence_msg.action} from #{presence_msg.client_id}"
-      channel.presence.get do |members|
-        puts "There are now #{members.length} clients present on this channel"
-      end
+    channel.presence.get do |members|
+      puts "There are #{members.length} clients present on this channel"
+      first = members[0]
+      puts "The first member is #{first.clientId} and their data is #{first.data}"
     end
   ```
 
-  "See this step in Github":https://github.com/ably/tutorials/commit/presence-ruby-step5
+As soon as you are attached to the channel, the presence set starts to sync with Ably. When you make this call to get the presence set, it will wait until it has finished syncing before completing. For small presence sets this will take only a few milliseconds; for larger ones (with hundreds of members) the full sync can take a little longer.
+
+blang[javascript].
+  Open this file in multiple browser tabs and inspect the browser's console. You should see new members appearing in the list of present members as you open more tabs and disappearing from the presence set when you close a tab.
+
+blang[nodejs].
+  Run your code with @node example.js@ in multiple terminal windows and inspect the output. You should see new members appearing in the list of present members when you run your app in a new terminal window. They will disappear from the presence set when you terminate the running program in a terminal window.
+
+blang[ruby].
+  Run your code with @ruby example.rb@ in multiple terminal windows and inspect the output. You should see new members appearing in the list of present members when you run your app in a new terminal window. They will disappear from the presence set when you terminate the running program in a terminal window.
+
+h2. Step 6 - Conclusion
+
+In this tutorial, you learned how to:
 
+* Attach to a channel
+* Enter (and leave) the presence set for the channel
+* Access the list of members present on a channel
+* Update a present member's status
+* Listen for presence changes
 
-And that's it. To see this in action, try out the live demo below.
+To see another example of Presence in action, try out the live demo below.
 
 h2(#live-demo). Live demo
```

Comment:
I'm not sure if we have an official stance on this, but personally I always like to have a demo (and link to the source) at a start of a tutorial when I'm going through. I find it easier to follow when I've an idea of how the end product will look as well as having the code to hand from the start if needed.

Reply:
Live demo removed, because it's not representative of what the tutorial teaches people to build.